mysql-test/main/type_blob.test

```diff
@@ -808,3 +808,54 @@ DELIMITER ;$$
 --source unusable_keys_joins.inc
 DROP TABLE t1;
 SET note_verbosity=DEFAULT;
+
+--echo #
+--echo # MDEV-36235 Incorrect result for BETWEEN over unique blob prefix
+--echo #
+# myisam has a special optimization for tables with one row
+CREATE TABLE t1 (c1 TINYBLOB, UNIQUE (c1(2))) engine=myisam;
+INSERT INTO t1 (c1) VALUES (1);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+DROP TABLE t1;
+
+CREATE TABLE t1 (c1 TINYBLOB, UNIQUE (c1(2))) engine=myisam;
+INSERT INTO t1 (c1) VALUES (1),(2),(3),(4),(5);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+DROP TABLE t1;
+
+CREATE TABLE t1 (c1 TINYBLOB, UNIQUE (c1(2)));
+INSERT INTO t1 (c1) VALUES (1),(2),(3),(4),(5);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+SELECT c1 FROM t1 WHERE 3 BETWEEN 10*POW(-1,c1) AND (c1);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 10*POW(-1,c1) AND (c1);
+DROP TABLE t1;
+
+CREATE TABLE t1 (c1 TINYBLOB NOT NULL);
+INSERT INTO t1 (c1) VALUES (1),(2),(3),(4),(5);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+DROP TABLE t1;
+
+CREATE TABLE t1 (c1 TINYBLOB, UNIQUE (c1(2))) engine=innodb;
+INSERT INTO t1 (c1) VALUES (1),(2),(3),(4),(5);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+ALTER TABLE t1 engine=myisam;
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+DROP TABLE t1;
+
+CREATE TABLE t1 (c1 TINYBLOB, UNIQUE (c1)) engine=innodb;
+INSERT INTO t1 (c1) VALUES (1),(2),(3),(4),(5);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+DROP TABLE t1;
+
+CREATE TABLE t1 (c1 VARBINARY(10), UNIQUE (c1));
+INSERT INTO t1 (c1) VALUES (1),(2);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+SELECT c1 FROM t1 IGNORE KEY(c1) WHERE 'a' BETWEEN 0 AND (c1);
+DROP TABLE t1;
+
+CREATE TABLE t1 (c1 BINARY(16), UNIQUE (c1));
+INSERT INTO t1 (c1) VALUES (1);
+INSERT INTO t1 (c1) VALUES (2);
+SELECT c1 FROM t1 WHERE 'a' BETWEEN 0 AND (c1);
+SELECT c1 FROM t1 IGNORE KEY(c1) WHERE 'a' BETWEEN 0 AND (c1);
+DROP TABLE t1;
```

Comment:
There is a file `type_binary.test` for tests with the `BINARY` data type. This chunk should probably go there.
I also suggest to move the previous chunk for `VARBINARY` to a new file `type_varbinary.test`